app/src/debug/java/timur/gilfanov/messenger/debug/DebugDataRepositoryImpl.kt

```diff
@@ -0,0 +1,278 @@
+                logger.w(TAG, "Invalid saved scenario: $scenarioName", e)
+                null
+            }
+        }
+
+    /**
+     * Toggle auto-activity feature
+     */
+    override suspend fun setAutoActivity(enabled: Boolean) {
+        updateSettings { current ->
+            current.copy(autoActivity = enabled)
+        }
+    }
+
+    /**
+     * Toggle debug notification visibility
+     */
+    override suspend fun setNotification(show: Boolean) {
+        updateSettings { current ->
+            current.copy(showNotification = show)
+        }
+    }
+
+    private fun startAutoActivity() {
+        stopAutoActivity() // Stop any existing activity
+
+        logger.d(TAG, "Starting auto-activity simulation")
+        autoActivityJob = coroutineScope.launch {
+            while (isActive) {
+                try {
+                    // Random delay between 5-15 seconds
+                    delay(Random.nextLong(AUTO_ACTIVITY_MIN_DELAY_MS, AUTO_ACTIVITY_MAX_DELAY_MS))
+                    simulateNewMessage()
+                } catch (e: CancellationException) {
+                    throw e // Re-throw cancellation
+                } catch (e: IllegalStateException) {
+                    logger.w(TAG, "Error in auto-activity simulation - illegal state", e)
+                } catch (e: IllegalArgumentException) {
+                    logger.w(TAG, "Error in auto-activity simulation - invalid arguments", e)
+                }
+            }
+        }
+    }
+
+    private fun stopAutoActivity() {
+        autoActivityJob?.cancel()
+        autoActivityJob = null
+        logger.d(TAG, "Stopped auto-activity simulation")
+    }
+
+    private fun simulateNewMessage() {
+        try {
+            // Get a random existing chat and add a message to it
+            val currentChats = remoteDebugDataSource.getChats()
+            if (currentChats.isNotEmpty()) {
+                val randomChat = currentChats.random()
+                val newMessage = sampleDataProvider.generateSingleMessage(randomChat)
+
+                logger.d(TAG, "Simulating new message in chat: ${randomChat.name}")
+                remoteDebugDataSource.addMessage(newMessage)
+            }
+        } catch (e: CancellationException) {
+            throw e // Re-throw cancellation
+        } catch (e: IllegalStateException) {
+            logger.w(TAG, "Failed to simulate new message - illegal state", e)
+        } catch (e: IllegalArgumentException) {
+            logger.w(TAG, "Failed to simulate new message - invalid arguments", e)
+        }
+    }
+
+    /**
+     * Fix message recipient IDs to match the chat ID
+     * This is necessary because SampleDataProvider generates random IDs for recipients
+     */
+    private fun fixMessageRecipients(chat: Chat): Chat {
+        val fixedMessages = chat.messages.map { message ->
+            when (message) {
+                is timur.gilfanov.messenger.domain.entity.message.TextMessage -> {
```

Comment:
Must import path to the class to have a consice code:
```suggestion
                is TextMessage -> {
```